### ISSUE TITLE
AndroidJavaClient exception

### ISSUE THREAD

**Issue by pavelveselev** [aspnet/AspNetCore#9862]
I ran AndroidJavaClient sample, and dot an Exception:

2019-04-25 14:01:42.877 15072-15328/com.microsoft.aspnet.signalr.samples.androidjavaclient E/OkHttpWebSocketWrapper: WebSocket closed from an error: Expected HTTP 101 response but was '200 OK'.
2019-04-25 14:01:42.878 15072-15328/com.microsoft.aspnet.signalr.samples.androidjavaclient I/WebSocketTransport: WebSocket connection stopping with code null and reason 'Expected HTTP 101 response but was '200 OK''.
2019-04-25 14:01:42.878 15072-15328/com.microsoft.aspnet.signalr.samples.androidjavaclient E/c*.m*.s*.HubConnection: HubConnection disconnected with an error Expected HTTP 101 response but was '200 OK'.
    
```
2019-04-25 14:01:42.885 15072-15328/com.microsoft.aspnet.signalr.samples.androidjavaclient E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: com.microsoft.aspnet.signalr.samples.androidjavaclient, PID: 15072
    java.lang.NullPointerException: Attempt to invoke virtual method 'void com.microsoft.signalr.HubConnection$ConnectionState.cancelOutstandingInvocations(java.lang.Exception)' on a null object reference
        at com.microsoft.signalr.HubConnection.stopConnection(HubConnection.java:431)
        at com.microsoft.signalr.HubConnection.lambda$start$6$HubConnection(HubConnection.java:301)
        at com.microsoft.signalr.HubConnection$$Lambda$19.invoke(Unknown Source:2)
        at com.microsoft.signalr.WebSocketTransport.onClose(WebSocketTransport.java:93)
        at com.microsoft.signalr.WebSocketTransport.lambda$start$1$WebSocketTransport(WebSocketTransport.java:56)
        at com.microsoft.signalr.WebSocketTransport$$Lambda$1.invoke(Unknown Source:2)
        at com.microsoft.signalr.OkHttpWebSocketWrapper$SignalRWebSocketListener.onFailure(OkHttpWebSocketWrapper.java:98)
        at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:570)
        at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:197)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
```
I've created simple Xamarin Android project, added this code:
```
var connection = new HubConnectionBuilder().WithUrl("https://my.site.com").Build();
connection.On<string>("SendInfo", x => Log.Info("SIGNALR", x));
connection.StartAsync();
```
and everything works fine. How to run AndroidJavaClient correctly?

**Comment by mikaelm12** [comment]
Can you provide more information about your environment? What API level are you targeting? 
Is the error just happening on start up or are you able to do anything before the exception? Uploading a repro app to github would be helpful

**Comment by pavelveselev** [comment]
Repo on GitHub: https://github.com/pavelveselev/SignalRSample
I published simple SignalR server to MS Azure: https://signalrserver20190507124505.azurewebsites.net/test

**Comment by mikaelm12** [comment]
Are you able to get the Xamarin client work with this? It seems like you don't have any hub methods on your server? 

The logs show that the issue starts when you get a message coming in. Theres a class cast exception in the logs which hint at the fact that whats coming isn't matching what we expect it to be.  Are you ever successfully processing the Handshake message?

**Comment by pavelveselev** [comment]
@mikaelm12 Xamarin client works with this. There's no any hub methods on my server. Server just sends broadcast messages (type String) to all clients every second. 
I've checked again - my server sends messages "SendTest" of type System.String. Android client expects to receive message "SendTest" of type kotlin.String. 
```
hubConnection?.on("SendTest", { message -> Log.i("SIGNALR!", message)}, String::class.java)
```
Is there a problem to cast System.String to kotlin.String?

**Comment by pavelveselev** [comment]
I've fixed the problem in my project (commit [7e4180af83e8af03ba3b798b6641620f55d20e5b):](https://github.com/pavelveselev/SignalRSample/commit/7e4180af83e8af03ba3b798b6641620f55d20e5b)
added to build.gradle:
```
    compileOptions {
        targetCompatibility = 1.8
        sourceCompatibility = 1.8
    }
```
Please add it to [docs](https://docs.microsoft.com/en-us/aspnet/core/signalr/java-client)

**Comment by pavelveselev** [comment]
@anurse please close the issue
